### ISSUE TITLE
False positive in MultiAliasImportRequireUse

### ISSUE THREAD

**Issue by randycoulman** [rrrene/credo#441]
### Precheck

* For proposing a new feature, be sure to name your issue "Proposal: ..."
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): 0.8.6
* Erlang/Elixir version (`elixir -v`): Erlang/OTP 20, Elixir 1.5.1
* Operating system: macOS 10.12.6

### What were you trying to do?

When attempting to upgrade credo from 0.8.1 to 0.8.6, I started seeing a new consistency check failure from the MultiAliasImportRequireUse rule.

The file in question defines a family of related structs.  Here's a skeleton of the relevant parts:

```elixir
defmodule InvoiceTracker.TimeSummary do
  alias InvoiceTracker.{ProjectTimeSummary, Rounding, TimeEntry}
  alias Timex.Duration
  #...
end

defmodule InvoiceTracker.ProjectTimeSummary do
  alias InvoiceTracker.{Detail, Rounding, TimeEntry}
  alias Timex.Duration
  # ...
end

defmodule InvoiceTracker.Detail do
  alias InvoiceTracker.{Rounding, TimeEntry}
  alias Timex.Duration
  # ...
end
```

The entire file is [here](https://github.com/randycoulman/invoice_tracker/blob/master/lib/invoice_tracker/time_summary.ex).

### Expected outcome

This is the output I get with credo 0.8.1:

```
Checking 12 source files ...

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.4 seconds (0.01s to load, 0.3s running checks)
```

### Actual outcome

This is the output I get with credo 0.8.6:

```
  Consistency
┃
┃ [C] ↗ Most of the time you are using the multi-alias/require/import/use
┃       syntax, but here you are using multiple single directives
┃       lib/invoice_tracker/time_summary.ex:100 #(InvoiceTracker.Detail)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.3 seconds (0.02s to load, 0.3s running checks)
```

I can make the warning go away by changing all three `alias Timex.Duration` lines to `alias Timex.{Duration}`, but I shouldn't have to do that for a single alias from that module.  I don't have to make this same change anywhere else in my project where I have this same pattern in single modules.  Because this file defines three modules, I'm guessing that's what triggers the issue.

**Comment by rrrene** [comment]
Thx for reporting this! :+1:

**Comment by tramjoe** [comment]
Adding to the same issue, another use case where using multi-alias most of the time ends up yielding false positives of multiple single directives is inside macros. Typically, here is a basic Phoenix endpoint definition that demonstrates the issue. Note that the multi-aliases in this source were added manually, as the Phoenix blueprint normally uses multi-line.

> 
>   Consistency                                                                   
> ┃ 
> ┃ [C] ↗ Most of the time you are using the multi-alias/require/import/use 
> ┃       syntax, but here you are using multiple single directives
> ┃       apps/register_web/lib/register_web.ex:33 #(RegisterWeb.channel)
> ┃ [C] ↗ Most of the time you are using the multi-alias/require/import/use 
> ┃       syntax, but here you are using multiple single directives
> ┃       apps/register_web/lib/register_web.ex:26 #(RegisterWeb.router)


```
defmodule RegisterWeb do
  @moduledoc false

  def controller do
    quote do
      use Phoenix.Controller, namespace: RegisterWeb
      import Plug.Conn
      import RegisterWeb.{Router.Helpers, Gettext}
    end
  end

  def view do
    quote do
      use Phoenix.View, root: "lib/register_web/templates",
                        namespace: RegisterWeb

      # Import convenience functions from controllers
      import Phoenix.Controller, only: [get_flash: 2, view_module: 1]
      import RegisterWeb.{Router.Helpers, ErrorHelpers, Gettext}
    end
  end

  def router do
    quote do
      use Phoenix.Router
      import Plug.Conn
      import Phoenix.Controller
    end
  end

  def channel do
    quote do
      use Phoenix.Channel
      import RegisterWeb.Gettext
    end
  end

  defmacro __using__(which) when is_atom(which) do
    apply(__MODULE__, which, [])
  end
end
```

**Comment by russmatney** [comment]
Is there any progress toward resolving this? This happens fairly often – a common pattern in test files that are testing multiple modules with a `use` that happens to alias its own modules. If there's any direction toward a solution, I'm happy to take a look later this month.

**Comment by TheFirstAvenger** [comment]
The issue here is that Credo is treating the file as one module, and therefore aliases in different sub-modules are lumped together and seen as multi-line aliases. The workaround for the originally reported issue is to unify the aliases at the top of the file. I have submitted a PR to the repo with an example of this https://github.com/randycoulman/invoice_tracker/pull/11. My suggestion for the phoenix example is to exclude the file from the check, as you are not able to unify these aliases.

**Comment by rrrene** [comment]
Closing due to inactivity. Please re-open at your convenience :+1:
